### DISCUSSION TITLE
Asyncua Server - Source TimeStamp

### DISCUSSION THREAD

**Discussion by marcos-schmitz** [FreeOpcUa/opcua-asyncio#1202] [Q&A]
Hello

I need some help with Asyncua Server. How do I make it update the value of the Source TimeStamp when some value is changed in UA Expert, for example?

**Comment by AndreasHeine** [comment]
You need to pass a DataValue which containes the Timestamps a StatusCode and the value ( Variant with value and DataType and information if it is an Array or not )


```python
  dv = ua.DataValue(
          ua.Variant(
              100.0, 
              ua.VariantType.Double
          ),
          StatusCode_=ua.StatusCodes.Good,
          SourceTimestamp=datetime.datetime.utcnow(),
          ServerTimestamp=datetime.datetime.utcnow()
      )
```

![image](https://user-images.githubusercontent.com/56362817/218657880-6c94decc-c0a2-4c08-b8e4-ccadfce6120c.png)

**Reply by marcos-schmitz** [reply]
Hello, thanks for the informations, but I have solved my problem with this: 
Set ServerTimestamp when attributes/values are written #1135

**Reply by AndreasHeine** [reply]
ok! you asked for the sourceTimestamp!?

**Reply by marcos-schmitz** [reply]
Yes, but I applied the same technique to update both values.
